Add support for PPC64LE

kbrenneman opened the pull request:
This change adds support for PPC64 little-endian systems.

The first change is all the structural stuff -- the configure script, makefiles, and adding new source files. There's placeholder comments for where the assembly code and other architecture-specific stuff would need to go.

The remaining changes fill in the assembly code in each file.

aaronp24 commented on src/GLdispatch/vnd-glapi/entry_files.mk:
```diff
@@ -37,6 +37,18 @@ MAPI_GLDISPATCH_ENTRY_FILES = entry_aarch64_tsd.c
+
+if GLDISPATCH_TYPE_PPC64LE_TLS
+MAPI_GLDISPATCH_ENTRY_FILES = entry_ppc64le_tls.c
+MAPI_GLDISPATCH_ENTRY_FILES += entry_x86_64_common.c
```
It might be nice to rename this file at some point and make the ARM version the special case.

kbrenneman replied:
That's true, the "x86_64" part isn't accurate anymore. We've already got an entry_common.c file, though, which would otherwise be the obvious choice.

Maybe something like entry_simple_common.c or entry_simple_asm.c?

aaronp24 replied:
Literally anything more generic than `x86_64` would be fine with me. :)

aaronp24 commented:
Ugh, reviewing one commit approves all of them? Whoops.

aaronp24 commented on src/GLdispatch/vnd-glapi/entry_ppc64le_tsd.c:
```diff
@@ -35,24 +35,68 @@
+    "  addi   1, 1, 144\n\t"                            \
+    "  ld     0, 16(1)\n\t"                             \
+    "  mtlr   0\n\t"                                    \
+    "  b      1050b\n"
```
As usual, I don't feel qualified to review these but I'll take your word for it that they work.

kbrenneman commented:
Okay, I added one more change to fix the filename: entry_x86_64_common.c is now entry_simple_asm.c.

I also added that file to the aarch64 build, since it doesn't need any special handling like ARMv7 does, and the functions in entry_aarch64_tsd.c were identical.
